kytos/lib/helpers.py

```diff
@@ -0,0 +1,113 @@
+    interface.id = "{}:{}".format(switch.dpid, port_number)
+    interface.name = name
+    interface.port_number = port_number
+    interface.switch = switch
+    interface.address = "00:00:00:00:00:00"
+    interface.lldp = True
+    return interface
+
+
+def get_link_mock(endpoint_a, endpoint_b):
+    """Return a link mock."""
+    link = create_autospec(Link)
+    link.endpoint_a = endpoint_a
+    link.endpoint_b = endpoint_b
+    link.metadata = {"A": 0}
+    return link
+
+
+def get_switch_mock(dpid, of_version=None):
+    """Return a switch mock."""
+    switch = create_autospec(Switch)
+    switch.dpid = dpid
+    if of_version:
+        switch.ofp_version = '0x0' + str(of_version)
+        switch.connection = get_connection_mock(of_version, switch)
+    return switch
+
+
+def get_connection_mock(of_version, switch, state=ConnectionState.NEW):
+    """Return a connection mock."""
+    protocol = create_autospec(ConnectionProtocol)
+    protocol.version = of_version
+    connection = create_autospec(Connection)
+    connection.protocol = protocol
+    connection.switch = switch
+    connection.address = "00:00:00:00:00:00"
+    connection.state = state
+    return connection
+
+
+def get_kytos_event_mock(name, content):
+    """Return a kytos event mock."""
+    event = create_autospec(KytosEvent)
+    event.name = name
+    event.content = content
+    event.message = content.get('message')
+    event.destination = content.get('destination')
+    event.source = content.get('source')
+    return event
+
+
+def get_topology_mock():
+    """Create a default topology."""
+    switch_a = get_switch_mock("00:00:00:00:00:00:00:01", 0x04)
+    switch_b = get_switch_mock("00:00:00:00:00:00:00:02", 0x04)
+    switch_c = get_switch_mock("00:00:00:00:00:00:00:03", 0x01)
+
+    interface_a1 = get_interface_mock("s1-eth1", 1, switch_a)
+    interface_a2 = get_interface_mock("s1-eth2", 2, switch_a)
+
+    interface_b1 = get_interface_mock("s2-eth1", 1, switch_b)
+    interface_b2 = get_interface_mock("s2-eth2", 2, switch_b)
+
+    interface_c1 = get_interface_mock("s3-eth1", 1, switch_c)
+    interface_c2 = get_interface_mock("s3-eth2", 2, switch_c)
+
+    switch_a.interfaces = {interface_a1.id: interface_a1,
+                           interface_a2.id: interface_a2}
+    switch_b.interfaces = {interface_b1.id: interface_b1,
+                           interface_b2.id: interface_b2}
+    switch_c.interfaces = {interface_c1.id: interface_c1,
+                           interface_c2.id: interface_c2}
+
+    link_1 = get_link_mock(interface_a1, interface_b1)
+    link_2 = get_link_mock(interface_a2, interface_c1)
+    link_3 = get_link_mock(interface_b2, interface_c2)
+
+    topology = MagicMock()
+    topology.links = {"1": link_1, "2": link_2, "3": link_3}
+    topology.switches = {switch_a.dpid: switch_a,
+                         switch_b.dpid: switch_b,
+                         switch_c.dpid: switch_c}
+    return topology
```

Comment:
I'm not sure if putting this mock here is correct, since `Topology` is an entity unknown to Kytos Core.

Reply:
Sure @ajoaoff, maybe we can put this mock only in NApps helpers, I'll move it. Thanks!